site/src/sphinx/advanced-structured-logging.rst

```diff
@@ -148,6 +152,68 @@ automatically for you:
         }
     }
 
+Enabling content previews
+-------------------------
+Armeria provides the ``requestContentPreview`` and ``responseContentPreview`` properties in :api:`RequestLog`
+to retrieve the textual representation of the first N bytes of the request and response content.
+However, the properties are disabled by default due to performance overhead and thus they always return ``null``.
+You can enable it when you configure :api:`Server`, :api:`VirtualHost` or :api:`Client`.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.server.ServerBuilder;
+    import com.linecorp.armeria.server.VirtualHostBuilder;
+
+    ServerBuilder sb = new ServerBuilder();
+    ...
+    // Enable previewing the content with the maxium length of 100 for textual content.
+    sb.contentPreview(100);
+    ...
+    VirtualHostBuilder vhb = new VirtualHostBuilder("http://example.com");
+    // In this case, the property of virtual host takes precedence over that of server.
+    vhb.contentPreview(150);
+    ...
+    sb.virtualHost(vhb.build());
+
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.ClientBuilder;
+    import com.linecorp.armeria.client.HttpClientBuilder;
+
+    ClientBuilder cb = new HttpClientBuilder();
+    ...
+    cb.contentPreview(100);
+
+Note that the contentPreview() method enables the previews only for textual content
+which meets the following cases:
+
+- when its type matches ``text/*`` or ``application/x-www-form-urlencoded``.
+- when its charset has been specified. e.g. application/json; charset=utf-8.
+- when its subtype is ``xml`` or ``json``. e.g. application/xml, application/json.
+- when its subtype ends with ``+xml`` or ``+json``. e.g. application/atom+xml, application/hal+json
+
+You can also use your own way to make the previews by customizing :api:`ContentPreviewerFactory`.
+The factory creates a :api:`ContentPreviewer` based on HTTP headers to produce the previews
+when a request or response ends.
+The following example enables the hex dump preview of first 100 bytes, regardless of the content type:
+
+.. code-block:: java
+
+    import io.netty.buffer.ByteBufUtil;
+    import com.linecorp.armeria.common.logging.ContentPreviewer;
+
+    ServerBuilder sb = new ServerBuilder();
+
+    // A user can use their customized previewer factory.
+    sb.contentPreviewerFactory((ctx, headers) -> {
+        // Produces the hex dump of the first 100 bytes.
+        return ContentPreviewer.ofBinary(100, byteBuf -> {
+            // byteBuf has no more than 100 bytes.
+            return ByteBufUtil.hexDump(byteBuf);
+        });
+    });
+
```

Comment:
And then, at this point, we could explain how to implement a custom `ContentPreviewer`, like which method is invoked when, with an actual working example of `ContentPreviewer` implementation.